adapters/repos/db/lsmkv/metrics.go

```diff
@@ -200,6 +207,18 @@ func (m *Metrics) MemtableWriteObserver(strategy, op string) BytesObserver {
 	}
 }
 
+func (m *Metrics) ReadObserver(op string) BytesReadObserver {
+	if m == nil {
+		return noOpNsReadObserver
+	}
```

Comment:
Wouldn't it be a "nil pointer dereference" if someone tried to call `m.ReadObserver()` on a nil-Metrics?